### ISSUE TITLE
Using php "declare" directive breaks page content in Packages section

### ISSUE THREAD

**Issue by outcomer** [phpDocumentor/phpDocumentor#2703]
## Expected behavior
If you start the class file like this
```
<?php
/**
 * Object
 *
 * @package SQL\Helpers
 */

declare(strict_types = 1);

namespace Smartbits\Messia\Helpers;

/**
 * This class contains static specific methods
 * to use as helpers without class instantiate.
 */
class Messia_Ext_Funcs {
    .....
}
```
you expect to see it in proper section on generated site pages here
![image](https://user-images.githubusercontent.com/40342952/100391673-7b4cde80-3034-11eb-860b-4c0c0ad5677d.png)

## Actual behavior
But in fact there will be nothing about this class in section `Packages` of docs home page.
![image](https://user-images.githubusercontent.com/40342952/100391898-35dce100-3035-11eb-978b-eef35932fb22.png)

## Steps to reproduce the problem
Just add at the beginning of any class PHP 7.* directive `declare(strict_types = 1);` and you will get described bug.

## Your environment
* Version used: 3.0.0
* Install method: via PHAR
* PHP version: 7.4
* Operating system: Windows 10

**Comment by jaapio** [comment]
I can confirm this is a bug. 

Files do not have comments. This is where things are getting messed up. The File docbloc belongs to the first statement in the file. Which means in this case `declare(strict_types = 1);` has a docblock.  Since we are ignoring this statement, accually we are ignoring all statements except some (functions, classes, interfaces, traits) 

The namespace has a special case, I think that needs to be extended. with all other elements.

**Comment by jaapio** [comment]
Possible patch, which needs to be checked and tested. 
First test did work, but we have to make sure we are not breaking things.

```
/**
 * Strategy to create File element from the provided filename.
 * This class supports extra middle wares to add extra steps to the creation process.
 */
final class File extends AbstractFactory
{
    /* ... */
    protected function createFileDocBlock(
        ?Context $context = null,
        array $nodes = []
    ) : ?DocBlockInstance {
        $node = null;
        foreach ($nodes as $n) {
            if (!in_array(get_class($n), self::SKIPPED_NODE_TYPES)) {
                $node = $n;
                break;
            }

            +if ($n->hasAttribute('comments')) {
             +   $node = $n;
             +   break;
            +}
        }
        
            /* ... */
```

**Comment by jaapio** [comment]
This issue has been solved in phpDocumentor v3.5
